File: Microsoft.CodeAnalysis.VisualBasic/Microsoft.CodeAnalysis.VisualBasic.Symbols/AttributeDataExtensions.cs
using System.Collections.Immutable;
using Microsoft.VisualBasic.CompilerServices;

namespace Microsoft.CodeAnalysis.VisualBasic.Symbols
{
	[StandardModule]
	internal sealed class AttributeDataExtensions
	{
		public static int IndexOfAttribute(this ImmutableArray<VisualBasicAttributeData> attributes, Symbol targetSymbol, AttributeDescription description)
		{
			int num = attributes.Length - 1;
			for (int i = 0; i <= num; i++)
			{
				if (attributes[i].IsTargetAttribute(targetSymbol, description))
				{
					return i;
				}
			}
			return -1;
		}
	}
}

File: Microsoft.CodeAnalysis.VisualBasic/Microsoft.CodeAnalysis/VisualBasic/Symbols/AttributeDataExtensions.vb
Imports Microsoft.CodeAnalysis
Imports Microsoft.CodeAnalysis.VisualBasic
Imports Microsoft.VisualBasic.CompilerServices
Imports System
Imports System.Collections.Immutable
Imports System.Runtime.CompilerServices

Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols
	Friend Module AttributeDataExtensions
		<Extension>
		Public Function IndexOfAttribute(ByVal attributes As ImmutableArray(Of VisualBasicAttributeData), ByVal targetSymbol As Symbol, ByVal description As AttributeDescription) As Integer
			Dim num As Integer
			Dim length As Integer = attributes.Length - 1
			Dim num1 As Integer = 0
			While True
				If (num1 > length) Then
					num = -1
					Exit While
				ElseIf (Not attributes(num1).IsTargetAttribute(targetSymbol, description)) Then
					num1 = num1 + 1
				Else
					num = num1
					Exit While
				End If
			End While
			Return num
		End Function
	End Module
End Namespace